go/internal/feast/server/http_server.go

```diff
@@ -192,22 +221,86 @@ func (s *httpServer) getOnlineFeatures(w http.ResponseWriter, r *http.Request) {
 		writeJSONError(w, fmt.Errorf("Error decoding JSON request data: %+v", err), http.StatusInternalServerError)
 		return
 	}
+
 	var featureService *model.FeatureService
+	var entitiesProto = make(map[string]*prototypes.RepeatedValue)
+	var requestContextProto = make(map[string]*prototypes.RepeatedValue)
+	var fVList = make([]string, 0)
+
 	if request.FeatureService != nil {
 		featureService, err = s.fs.GetFeatureService(*request.FeatureService)
 		if err != nil {
 			logSpanContext.Error().Err(err).Msg("Error getting feature service from registry")
 			writeJSONError(w, fmt.Errorf("Error getting feature service from registry: %+v", err), http.StatusInternalServerError)
 			return
 		}
+		for _, fv := range featureService.Projections {
+			fVList = append(fVList, fv.Name)
+		}
+	} else if request.Features != nil && len(request.Features) > 0 {
+		log.Info().Msgf("request.Features %v", request.Features)
+		for _, featureName := range request.Features {
+			fvName, _, err := onlineserving.ParseFeatureReference(featureName)
+			if err != nil {
+				logSpanContext.Error().Err(err)
+				writeJSONError(w, fmt.Errorf("Error parsing feature reference %s", featureName), http.StatusBadRequest)
+				return
+			}
+			fVList = append(fVList, fvName)
+		}
+	} else {
+		logSpanContext.Error().Msg("No feature views or feature services specified in your request")
+		writeJSONError(w, errors.New("No feature views or feature services specified in your request"), http.StatusBadRequest)
+		return
+	}
+	requestSources, err2 := s.fs.GetRequestSources(fVList)
```

Comment:
With the way I've setup the checks, I had to make the calls twice, once for feature services and one for feature views. I thought of making a single call to avoid multiple calls.